coap-gateway/service/clientObserveHandler.go

```diff
@@ -114,8 +136,47 @@ func (s *resourceSubscription) eventHandler(e *pb.Event) error {
 	return nil
 }
 
-func (s *resourceSubscription) Init() error {
-	return s.sub.Init(s.client.server.ownerCache)
+func (s *resourceSubscription) eventHandlerSync(e *pb.Event) error {
```

Comment:
Doesn't seem necessary now, just call `s.eventHandler(e)` instead of `s.eventHandlerSync(e)`.